datalad/support/annexrepo.py

```diff
@@ -1620,7 +1621,8 @@ def _check_files(self, fn, files, batch):
         # reports a type change as modified.
         modified = [
             f for f in self.call_git_items_(
-                ['diff', '--name-only', '-z'], sep='\0')
+                ['diff', '--name-only', '-z'], sep='\0',
```

Comment:
As mentioned in my other response, I think `git-diff` is an example of a command that may refresh the index and is not safe to mark as `read_only=True`, provided that we're using `read_only` to decide whether to take a lock for parallel operations.


```suggestion
                ['diff-files', '--name-only', '-z'], sep='\0',
```